lib/sshkit/backends/connection_pool.rb

```diff
@@ -61,9 +61,6 @@ def with(connection_factory, *args)
     yield(conn)
   ensure
     cache.push(conn) unless conn.nil?
-    # Sometimes the args mutate as a result of opening a connection. In this
-    # case we need to update the cache key to match the new args.
-    update_key_if_args_changed(cache, args)
```

Comment:
This is not necessary anymore because hostname and username don't mutate as a result of opening a connection.